src/core/server/saved_objects/export/get_sorted_objects_for_export.ts

```diff
@@ -94,6 +102,12 @@ async function fetchObjectsToExport({
     if (typeof search === 'string') {
       throw Boom.badRequest(`Can't specify both "search" and "objects" properties when exporting`);
     }
+    if (references && references.length) {
```

Comment:
Yea. Honestly the whole `exportSavedObjectsToStream` static function should be exposed from the SO client instead. That would avoid having to pass `savedObjectsClient` and `exportSizeLimit` as parameters, and allow client wrappers to enhance it properly.

Changing the signature of `fetchObjectsToExport ` to something like `options: ExportByType | ExportByObjects` is not that trivial also, as the main consumer is an http endpoint, so doing that would force to move the options validation and specialized option types construction down to it, so it would probably make sense to do both at the same time.

WDYT, should I open an issue for that?

Reply:
I will refactor `fetchObjectsToExport` to be 'split' into `fetchObjectsToExportByType` and `fetchObjectsToExportByObjects` in this PR to remove the introduced tech debt. Moving the method to the SOC is a more impactful change that is not currently driven by any real need.

Reply:
👍 for splitting the function in a follow-up. we can add a comment for further ourselves to consider moving this logic to SOC

Reply:
Started the work in https://github.com/elastic/kibana/pull/79096/commits/c057cd2c6ac80c942bcef3c611c084766dde857c. Further changes impacting the current consumers of `exportSavedObjectsToStream` will need to be as a follow-up